diff -ruN libvorbisidec-1.2.1.orig/Android.mk libvorbisidec-1.2.1/Android.mk
--- libvorbisidec-1.2.1.orig/Android.mk	1969-12-31 16:00:00.000000000 -0800
+++ libvorbisidec-1.2.1/Android.mk	2017-10-22 17:27:12.176911197 -0700
@@ -0,0 +1,32 @@
+LOCAL_PATH := $(call my-dir)
+
+include $(CLEAR_VARS)
+
+LOCAL_MODULE := vorbisidec
+
+LOCAL_C_INCLUDES :=
+
+LOCAL_CFLAGS :=
+
+ifeq ($(TARGET_ARCH),arm)
+	LOCAL_CFLAGS += -D_ARM_ASSEM_
+endif
+
+LOCAL_SRC_FILES += \
+    mdct.c.arm \
+    block.c \
+    window.c.arm \
+    synthesis.c \
+    info.c \
+    floor1.c.arm \
+    floor0.c.arm \
+    vorbisfile.c.arm \
+    res012.c \
+    mapping0.c \
+    registry.c \
+    codebook.c \
+    sharedbook.c.arm \
+
+LOCAL_SHARED_LIBRARIES := ogg
+
+include $(BUILD_STATIC_LIBRARY)
diff -ruN libvorbisidec-1.2.1.orig/misc.h libvorbisidec-1.2.1/misc.h
--- libvorbisidec-1.2.1.orig/misc.h	2017-10-22 17:26:08.725205666 -0700
+++ libvorbisidec-1.2.1/misc.h	2017-10-12 20:05:06.719770194 -0700
@@ -30,6 +30,7 @@
 
 #include "asm_arm.h"
 #include <stdlib.h> /* for abs() */
+#include <endian.h>
   
 #ifndef _V_WIDE_MATH
 #define _V_WIDE_MATH
